src/makeInfinityAdder.js

```diff
@@ -6,7 +6,34 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+  let sum = 0;
+
+  return function makeAdder(value1) {
+    if (!value1) {
+      if (result !== 0) {
+        sum = result;
+        result = 0;
+
+        return sum;
+      }
+
+      return result;
```

Comment:
1. you should check value for undefined. If it's true - reset result
2. if it's not true just add a value to result
3. `adder` function should return itself
```suggestion
    if (value === undefined) {
      const sum = result;
      result = 0;

      return sum;
    }
      
    result += num

    return adder;
```